core/src/main/java/org/kiwix/kiwixmobile/core/main/FileReader.kt

```diff
@@ -15,31 +15,19 @@
  * along with this program. If not, see <http://www.gnu.org/licenses/>.
  *
  */
-package org.kiwix.kiwixmobile.core.main;
-
-import android.content.Context;
-import java.io.BufferedReader;
-import java.io.InputStream;
-import java.io.InputStreamReader;
-
-public class FileReader {
-
-  public String readFile(String filePath, Context context) {
-    try {
-      StringBuilder buf = new StringBuilder();
-      InputStream json = context.getAssets().open(filePath);
-      BufferedReader in =
-        new BufferedReader(new InputStreamReader(json, "UTF-8"));
-      String str;
-
-      while ((str = in.readLine()) != null) {
-        buf.append(str);
-      }
-
-      in.close();
-      return buf.toString();
-    } catch (Exception e) {
-      return "";
+package org.kiwix.kiwixmobile.core.main
+import android.content.Context
+import java.io.IOException
+class FileReader {
+  fun readFile(filePath: String?, context: Context): String {
+    return try {
+      val buf = StringBuilder()
+      val json = context.assets.open(filePath)
+      json.bufferedReader().use(buf::append)
```

Comment:
This writes only the first line to the stringBuilder

Reply:
ooh! I should see the documentation before use it